### ISSUE TITLE
Questions about Pearson depth loss

### ISSUE THREAD

**Issue by coolball666** [ForMyCat/SparseGS#29]
Hi, thanks a lot for sharing your wonderful work, but I'm a little confused about the depth used for the Pearson loss.
The paper gives the formula of the softmax depth you have applied. It is the natural logarithm of some sort of absolute depth to my understanding.
The code of the rasterizer also confirms this.
https://github.com/ForMyCat/SparseGS/blob/95e7aef29c5562400d3b2b38cc7e90436a432b7c/submodules/diff-gaussian-rasterization-softmax/cuda_rasterizer/forward.cu#L415
And during the rendering process, the exponent of this rendered depth is used to visualize the depth result.
https://github.com/ForMyCat/SparseGS/blob/95e7aef29c5562400d3b2b38cc7e90436a432b7c/render.py#L43
So I believe the real relationship between the rendered depth and gt_depth is of the form $D_{\text{gt}}=Exp(D_{\text{softmax}})$

So things become very confusing to me. When calculating the Pearson loss in the training process, the exponent of the rendered depth is not used.
https://github.com/ForMyCat/SparseGS/blob/95e7aef29c5562400d3b2b38cc7e90436a432b7c/train.py#L149-L155
As for my knowledge, the following proposition holds for PCC:

$|PCC(X, Y)|=1\iff\mathbb{P}(Y=aX+b)=1$, where $a, b$ are constants and $P$ is probability.
The PCC seems cannot express the exponential relationship of the used depth and gt_depth in the computation of Pearson loss in the code.

So after taking the exponent of the softmax depth, I ran another training in my own dataset (a very large scene). The results improved quite a lot and the Pearson loss and local Pearson loss descended significantly, compared with the fluctuating Pearson loss before taking the exponent. So I wonder whether there are some mistakes in the code?

Thank you in advance for your reply.

**Comment by sairisheek** [comment]
Hi,

Thanks for your interest in our work.

The log is used to simplify backpropagation for the softmax depth by avoiding the quotient rule, which improves numerical stability (preventing blowup in $e^ {\beta w_i}$). We appreciate your feedback on the misalignment between the ground truth and softmax depth. Note that this is not the latest code; we have an option to load  log(gt_depth) and compute depth loss accordingly.

**Comment by coolball666** [comment]
I got it, thank you for your answer, and thanks again for your wonderful work!
